File: solutions/haskell/sum-of-multiples/1/src/SumOfMultiples.hs
module SumOfMultiples (sumOfMultiples) where
import Data.List

sumOfMultiples :: [Integer] -> Integer -> Integer
sumOfMultiples factors limit = sum $ map head $ group $ sort $ concatMap (\base -> (m base limit)) factors

m :: Integer -> Integer -> [Integer]
m 0 limit = [0]
m base limit = [x * base | x <- [0..(div (limit - 1) base)]]

File: solutions/kotlin/sum-of-multiples/2/src/main/kotlin/SumOfMultiples.kt
object SumOfMultiples {

    fun sum(factors: Set<Int>, limit: Int): Int {
        var ans = 0
        for (i in 1 until 20) {
            for (f in factors) {
                if (i % f == 0) {
                    ans += i
                    break
                }
            }
        }
        return ans
    }
}
